### DISCUSSION TITLE
Creation of folder on Github

### DISCUSSION THREAD

**Discussion by habibthadev** [community/community#51437] [Repositories]
### Select Topic Area

General

### Body

Hi everyone.

My name is Habib.
I'm a junior frontend developer and I'm currently having issue with creating/importing folders to GitHub.

N.B: I code with phone and I can't drag and drop the folder.

Please, suggest a solution to this

**Comment by rilysh** [comment]
@Habib-0007 you can manually create folders on GitHub.

1. Create or open an existing repository and click that drop down menu (stays at right)
2. Click on create new file
3. Write the folder name, you'd like to have there. For example write `myfolder/test` and GitHub will create a folder called `myfolder` and will create a file, called `test`. If you don't write the file name after the folder name, e.g. `test`, GitHub will normally create a file which will be called `myfolder`. So you need to always temporarily create a file.

### Images
![Screenshot_20230330-083914_Firefox](https://user-images.githubusercontent.com/71683721/228719312-d35ab4f2-a282-4fdf-9cb9-1a04975966a8.png)
![Screenshot_20230330-083933_Firefox](https://user-images.githubusercontent.com/71683721/228719323-6c32af26-7073-44c0-8247-ee5abd2a2be4.png)

For instance try steps I wrote. Create a new repo and create `myfolder/test`.

Note: You can replace `myfolder` and `test` with any other name.

**Comment by habibthadev** [comment]
Closed

**Comment by bronglil** [comment]
Using Git Command Line:
If you're comfortable with using the command line, you can follow these steps to create and push folders to GitHub:

a. Install a Git client on your phone if you haven't already. There are apps available for both Android and iOS.
b. Open the terminal or command prompt in the app.
c. Navigate to the directory where your project is located using the cd command.
d. Use standard Git commands to create folders and add files. For example:

`mkdir folder_name
git add .
git commit -m "Added a new folder"
git push origin main`

Using GitHub's Web Interface:
GitHub's web interface allows you to create and upload files and folders directly. Follow these steps:

a. Open the repository on GitHub in your mobile browser.
b. Tap the "Add file" dropdown, then choose "Upload files".
c. Tap "Choose your files" and select the folder and files you want to upload. You might need to compress the folder into a zip file before uploading.
d. Add a commit message and choose the branch (e.g., "main") to commit to.
e. Click the "Commit changes" button.
